### PR TITLE
[dbt] DbtProject apidoc

### PR DESCRIPTION
**Pull request by alangenfeld**
## How I Tested These Changes

make apidoc-build
make dev

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#21197** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21197?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ dagster-dbt
 ***********
 
 .. click:: dagster_dbt.cli.app:project_app_typer_click_object
-    :prog: dagster-dbt project scaffold
+    :prog: dagster-dbt project
     :nested:
```
Comment (rexledesma) [inline docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst:21]: Let's just add a new item for `dagster-dbt project prepare-for-deployment`.

It doesn't render properly when the full command is omitted: https://dagster-docs-qk9unul9n-elementl.vercel.app/_apidocs/libraries/dagster-dbt#dagster-dbt

File: docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst [hunk 2]
```diff
@@ -52,6 +48,8 @@ Assets (dbt Core)
 
 .. autofunction:: get_asset_keys_by_output_name_for_source
 
+.. autoclass:: DbtProject
```
Comment (rexledesma) [inline docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst:51]: Can we add some API examples + docstring to `DbtProject`? It looks barren rn: https://dagster-docs-qk9unul9n-elementl.vercel.app/_apidocs/libraries/dagster-dbt#dagster-dbt-project

**Review by rexledesma — CHANGES_REQUESTED** [review]
See comments

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 3]
```diff
@@ -74,6 +74,74 @@ def prepare(self, project: "DbtProject") -> None:
 
 @experimental
 class DbtProject(DagsterModel):
+    """Representation of a dbt project and related settings that assist with managing manifest.json preparation.
+
+    By default, using this helps achieve a setup where
+        * during development, reload the manifest at run time to pick up any changes.
+        * when deployed, expect a manifest that was created at build time to reduce start-up time.
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:81]: The formatting of bullets looks kinda weird: https://al-04-12--dbt-dbtproject-apidoc.dagster.dagster-docs.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DbtProject

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 4]
```diff
@@ -74,6 +74,74 @@ def prepare(self, project: "DbtProject") -> None:
...
+
+    The cli dagster-dbt project prepare-for-deployment can be used as part of the deployment process to
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:83]: And potentially link?

```suggestion
    The cli ``dagster-dbt project prepare-for-deployment`` can be used as part of the deployment process to
```

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 5]
```diff
@@ -74,6 +74,74 @@ def prepare(self, project: "DbtProject") -> None:
...
+    handle manifest.json preparation.
+
+    This object can be passed directly to DbtCliResource.
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:86]: Potentially link?

```suggestion
    This object can be passed directly to :py:class:`~dagster_dbt.DbtCliResource`.
```

File: docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst [hunk 6]
```diff
@@ -20,6 +20,10 @@ dagster-dbt
     :prog: dagster-dbt project scaffold
     :nested:
 
+.. click:: dagster_dbt.cli.app:project_app_typer_click_object
+    :prog: dagster-dbt project prepare-for-deployment
+    :nested:
```
Comment (rexledesma) [inline docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst:25]: This is rendering a bit weird, I wonder if we should get rid of `:nested:`, or maybe add some docstring

**Review by rexledesma — APPROVED** [review]
More nit comments, but should be fine with some tweaks

File: docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst (hunk 6 above)
Comment (alangenfeld) [reply docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst:25]: hmm, is there just something up with my local apidoc build artifacts or something? This looks good https://docs.dagster.io/_apidocs/libraries/dagster-dbt

Comment (alangenfeld) [reply docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst:25]: I keep getting sphinx errors if i try to set `full` on the `prepare-for-deployment` entry (it works on `scaffold`) no matter what i do with the doc strings, did you get the api doc build to work?

Comment (rexledesma) [reply docs/sphinx/sections/api/apidocs/libraries/dagster-dbt.rst:25]: yeah I got it to work after I removed all the backticks in `app.py`

**Pull request merged**